### PR TITLE
Honor origin's HTTP status code

### PR DESCRIPTION
**Pull request by yoya**
This PR resolve issue https://github.com/smartnews/yoya-thumber/issues/9

In an environment with CDN middleboxes, if yoya-thumber receive StatusNotFound(404) from origin and send StatusBadGateway(502) to the client, the contents will not be cached on CDN and cause a load problem.

### CODE REVIEW

**Comment by yoya** [comment]
With this fix applied, getting a 302 Found status code from the origin doesn't seem to send the URI to the client.
A common problem with reverse proxies is that if the origin URI is relayed to the client as is, the client may bypass the proxy (yoya-thumber) and communicate directly with the origin.
That shouldn't be yoya-thumber's intent.

**Comment by yoya** [comment]
Refine the treatment of HTTP status code. 

 origin     client
- 200     => 200 (StatusOK)
- 4xx,5xx => 4xx,5xx (status unchanged) *Fix by this PR*
- other   => 502 (BadGateway)

bad parameter on URL => 400 (BadRequest)

PR: https://github.com/yoya/yoya-thumber/commit/ec085ed4972c8209d98beddf3663a31e33ef9b2d

File: thumberd/thumberd.go [hunk 1]
```diff
@@ -187,14 +187,21 @@ func myClientImageGet(imageUrl string, referer string, userAgent string) (*http.
...
+	if srcReader.StatusCode == http.StatusOK {
+		return srcReader, nil, http.StatusOK // SUCCESS
+	}
+	srcReader.Body.Close()
+	// In case of 4xx or 5xx, send status to the client unchanged.
+	if srcReader.StatusCode >= http.StatusBadRequest {
+		return nil, errors.New("upstream status:" + srcReader.Status), srcReader.StatusCode // FAILED
 	}
-	return srcReader, nil
+	// other status 1xx, 2xx(except for 200), 3xx,
```
Comment (simonwu-sn) [inline thumberd/thumberd.go:202]: Do we send a second request when 301, 302 returned?

And why we treat 1xx, 2xx(except for 200), 3xx. as Gateway exception

Comment (yoya) [reply thumberd/thumberd.go:202]: I've maintained the previous behavior, but I welcome the suggested improvements.
=> 301, 302

It indicates that yoya-thumber does not support it, but I would like to know if you have a better status code.
=> 1xx, 2xx(except for 200), 3xx. as Gateway exception

Comment (simonwu-sn) [reply thumberd/thumberd.go:202]: Looks good to me.

**Review by simonwu-sn — APPROVED** [review]

**Pull request merged**